src/server.ts

```diff
@@ -111,6 +111,19 @@ const serializeToHTML = (document: Document) => (doc: ProsemirrorNode) => {
   });
   const throwAwayDiv = document.createElement("div");
   throwAwayDiv.appendChild(serializedFragment);
+
+  // removes button and select html elements from code blocks 
+  // prevents these html elements from messing up flashcard and quiz generation
+  const codeChilds = throwAwayDiv.getElementsByClassName( 'code-block' );
+
+  Array.from( codeChilds ).forEach( ( child ) => {
+    const firstChild = child.firstChild;
+
+    if ( firstChild ) {
+      child.removeChild( firstChild );
+    }
+  } );
```

Comment:
do you think it's better to remove the nodes here, or to pass something like `softToDOM` and handle that in `CodeFence.js` i.e. similar to [this](https://github.com/Knowt/rich-markdown-editor/blob/2a0cdddcec32a2813e903c813f6e0d68f38ae794/src/nodes/Heading.ts#L74)

Reply:
Yeah it would be better this way actually for code consistency and performance as well -> good call imma make the change. Tbh I didn't know what `softToDOM` was used for before lol

Reply:
switched to softToDOM @hadielyakhni